list-sensors.cpp

```diff
@@ -354,6 +376,83 @@ struct CmpSensorsName
     }
 };
 
+using Path = std::string;
+using Service = std::string;
+using Interface = std::string;
+using Interfaces = std::vector<Interface>;
+using ObjectsMap = std::map<Service, Interfaces>;
+using Objects = std::map<Path, ObjectsMap, CmpSensorsName>;
+
+/**
+ * @brief Run infinite loop to print sensor values each \p watch_interval
+ * seconds
+ *
+ * @param watch_list - List of sensor names to print
+ * @param watch_interval - Interval to wait
+ * @param objects - List of all sensors in the system
+ * @return EXIT_FAILURE
+ */
+static int watch_senors(const std::vector<std::string>& watch_list,
+                        const int& watch_interval, const Objects& objects)
+{
+    std::vector<std::pair<Service, Path>> sensors;
+
+    // we want to display sensors in order they were specified by user, so we
+    // cant just loop over objects and then use find on sensors_list
+    for (const auto& name : watch_list)
+    {
+        bool found = false;
+        for (const auto& obj : objects)
+        {
+            size_t name_pos = obj.first.rfind('/');
+            if (name == obj.first.substr(name_pos + 1))
+            {
+                found = true;
+                for (const auto& service : obj.second)
+                {
+                    sensors.emplace_back(std::pair(service.first, obj.first));
```

Comment:
`std::pair` is excess here, `emplace_back` expects the arguments of item constructor. In your case, it's a copy that isn't really required.

Reply:
sorry, don't get your point here, using just
`sensors.emplace_back(service.first, obj.first);`
don't compiles.